.github/workflows/main.yml

```diff
@@ -78,6 +78,13 @@ jobs:
         run: |
           make crouton
 
+      - name: Update App Version
+        if: github.ref_type == 'tag'
+        run: >-
+          if [[ $GITHUB_REF =~ ^([0-9]+\.){2}(\*|[0-9]+)(-beta)?([0-9]{0,2})?$ ]]; then
```

Comment:
this matches
`d.d.d`
`d.d.d-beta`
`d.d.d-betad`

but not
`d.d.d-`
`d.d.d-d`
`d.d.d-blah`